pymongo/auth_oidc.py

```diff
@@ -84,6 +86,17 @@ def get_current_token(self, credentials, address, use_callbacks=True):
         if not use_callbacks and not current_valid_token:
             return None
 
+        # Ensure that the desired address is allowed.
+        found = False
+        allowed_hosts = credentials.mechanism_properties.allowed_hosts
+        for patt in allowed_hosts:
+            if fnmatch.fnmatch(address[0], patt):
+                found = True
+        if not found:
+            raise PyMongoError(
+                f"Refusing to connect to {address[0]}, which is not in authOIDCAllowedHosts: {allowed_hosts}"
```

Comment:
Would it be possible for us to make this parameter snake-case? it's really, really hard to parse `authOIDCAllowedHosts` visually.

Reply:
Renamed to `allowed_hosts` here and in the spec, and moved it to an authorization mechanism.